core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -372,8 +367,14 @@ private boolean manifestHasDeletedFiles(
 
     for (ManifestEntry<F> entry : reader.liveEntries()) {
       F file = entry.file();
+
+      // add path-based delete to set of files to be deleted
+      if (deletePaths.contains(CharSequenceWrapper.wrap(file.path()))) {
```

Comment:
Why do we actually add the file to the set? I worry that it becomes indistinguishable after that. I feel it would be more obvious to keep checking `deletePaths.contains(file.path())` in addition to checking the file set. If this is done purely to validate missing paths, I think it won't be enough.

Reply:
my goal was to reduce handling of paths vs files in a bunch of places and only deal with actual files and also reduce required error handling in `validateRequiredDeletes()` but as you mentioned in another comment this misses the case when a path-based delete is performed with a non-existing file and there wasn't any test coverage for this. I've added test coverage and rewrote this